### PR TITLE
School Library: setup associations

### PR DESCRIPTION
**Pull request by mahdinoori2000**
What is added

- [ ] Create a class Classroom with the following:
- @label instance variable initialized in the constructor.
- Setter and getter for @label.

- [ ] Create the has-many/belongs-to relationship between Classroom and Student. The following implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
- When adding a student to a classroom it also sets the classroom for the student.
- When setting the classroom for a student it also adds it to the classrooms' students.

- [ ] Create a class Book with the following:
- @title and @author instance variables initialized in the constructor.
- Setters and getters for instance variables.

- [ ] - Create a class Rental with the following:
- @date instance variable initialized in the constructor.
- Setter and getter for @date .

- [ ] Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require_relative 'person'
+require_relative 'rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (mukhammadrizooff) [inline book.rb]: - [ ] [REQUIRED] Good job so far! You should be created the many-to-many relationship between `Person` and `Book` using the intermediate class `Rental` like that:, they should be implemented similar to the following:
```ruby.rb
attr_accessor :title, :author, :rentals
```

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require_relative 'student'
+
+class Classroom
+  attr_accessor :lable, :students
```
Comment (mukhammadrizooff) [inline classroom.rb]: - [ ] [REQUIRED] Good job so far! Please implement `:students` accessor to `attr_reader` 🙏🏼 

```ruby
attr_reader :students
```

File: person.rb [hunk 3]
```diff
@@ -4,20 +4,25 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rental
```
Comment (mukhammadrizooff) [inline person.rb]: - [ ] [REQUIRED] Good job so far! Please implement `Getters` and `Setters` for `@rentals` should be present in `attr_accessor` 🙏🏼 

```ruby
attr_accessor :name, :age, :rentals
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @mahdinoori2000, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @mahdinoori2000 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Ruby features ✔️ 
- [x] Kudus on meeting the requirements suggested  ✔️



Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**